apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.Data/ExecuteHelper.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Google.Cloud.Spanner.V1.Logging;
+
+namespace Google.Cloud.Spanner
+{
+    internal static class ExecuteHelper
+    {
+        internal static void WithErrorTranslationAndProfiling(Action t, string name)
+        {
+            try
+            {
+                Stopwatch sw = null;
+                if (Logger.LogPerformanceTraces)
+                    sw = Stopwatch.StartNew();
+                t();
+                if (sw != null)
+                    Logger.LogPerformanceCounterFn($"{name}.Duration", x => sw.ElapsedMilliseconds);
+            }
+            catch (Exception e)
```

Comment:
I'll need to check the scope of out variables, but I believe we'll be able to rewrite this as:

```csharp
catch (Exception e) when (SpannerException.TryTranslateRpcException(e, out var translatedException))
{
    throw translatedException
}
```

Can't do it until we've merged the csproj changes though, of course.